server/lomas_server/worker.py

```
@@ -0,0 +1,272 @@
+        return CostResponse(epsilon=eps_cost, delta=delta_cost)
+    except KNOWN_EXCEPTIONS as exc:
+        known_exc = handle_known_exceptions(exc)
+        print(f" [-] KNOWN_EXCEPTIONS ({known_exc.status_code}|{known_exc.body})")
+        return known_exc.body, known_exc.status_code
+    except Exception as e:
+        return e
+
+
+def handle_query(body):
+    start_sec = time.time()
+    message = body.decode()
+    user_name, dp_library, query_json = message.split(":", 2)
+
+    match dp_library:
+        case DPLibraries.SMARTNOISE_SQL:
+            query_json = SmartnoiseSQLQueryModel.model_validate_json(query_json)
+
+        case DPLibraries.SMARTNOISE_SYNTH:
+            query_json = SmartnoiseSynthQueryModel.model_validate_json(query_json)
+
+        case DPLibraries.OPENDP:
+            query_json = OpenDPQueryModel.model_validate_json(query_json)
+
+        case DPLibraries.DIFFPRIVLIB:
+            query_json = DiffPrivLibQueryModel.model_validate_json(query_json)
+
+    try:
+        data_connector = data_connector_factory(
+            query_json.dataset_name,
+            admin_database,
+            private_credentials,
+        )
+
+        dp_querier = querier_factory(
+            dp_library,
+            data_connector=data_connector,
+            admin_database=admin_database,
+        )
+        query_response = dp_querier.handle_query(query_json, user_name)
+        elapsed = time.time() - start_sec
+        print(f" [x] Done ({elapsed:.2f})")
+        return query_response
+    except KNOWN_EXCEPTIONS as exc:
+        known_exc = handle_known_exceptions(exc)
+        print(f" [-] KNOWN_EXCEPTIONS ({known_exc.status_code}|{known_exc.body})")
+        return known_exc.body, known_exc.status_code
+    except Exception as e:
+        return e
+
+
+def handle_dummy_query(body):
+    start_sec = time.time()
+    message = body.decode()
+    user_name, dp_library, query_model = message.split(":", 2)
+
+    match dp_library:
```

Should we raise an exception if unknown library? I guess it is supposed to be impossible.

---

ah yes, you would just need to call `init_telemetry()` with the worker ressource and replace print with logging

and these 2 lines would add tracing for all queues:
```
from opentelemetry.instrumentation.aio_pika import AioPikaInstrumentor
AioPikaInstrumentor().instrument()
```
but can also be done later